toolbox/pdf: Handle special cases for PDF page download in double page view

stweil opened the pull request:
The old code handled the cases where either left and right page had a
PDF or where there was no PDF for both pages.

The new code also handles the cases where only one of both pages has a PDF.
It then shows the correct download link (either for the left side or for
the right side).

Signed-off-by: Stefan Weil sw@weilnetz.de

sebastian-meyer commented:
Short array notation ($a = [1, 2, 3, 4]) was added with PHP 5.4. Since Goobi.Presentation currently still supports PHP 5.3 (just like TYPO3 6.2 does), please change this accordingly.

stweil committed:
toolbox/pdf: Handle special cases for PDF page download in double page view

The old code handled the cases where either left and right page had a
PDF or where there was no PDF for both pages.

The new code also handles the cases where only one of both pages has a PDF.
It then shows the correct download link (either for the left side or for
the right side).

Signed-off-by: Stefan Weil <sw@weilnetz.de>

stweil commented:
@sebastian-meyer: In the updated patch the short array notation was replaced by the old array notation.

Additional remark: anybody who still uses PHP 5.3 should consider upgrading. Although TYPO3 formally supports 5.3, they suggest at least 5.4. As PHP 5.3 is no longer supported (since August 2014?), that suggestion is reasonable.

sebastian-meyer commented:
Thank you very much!

I agree with your recommendation to use PHP 5.4, but we shouldn't enforce it as long as TYPO3 doesn't.
